### DISCUSSION TITLE
Support for Bzlmod (new external dependency system of Bazel)

### DISCUSSION THREAD

**Discussion by luangong** [nlohmann/json#3916] [Ideas]
It would be very nice if we could support Bzlmod.

From the official documentation of Bazel:

> Bzlmod is the codename of the new [external dependency](https://bazel.build/docs/external) system introduced in Bazel 5.0. It was introduced to address several pain points of the old system that couldn't feasibly be fixed incrementally; see the [Problem Statement section of the original design doc](https://docs.google.com/document/d/1moQfNcEIttsk6vYanNKIy3ZuK53hQUFq1b1r0rmsYVg/edit#heading=h.xxnnwabymk1v) for more details.

I’m in the process of migrating to Bzlmod for my personal projects but many third-party dependencies still don’t support Bzlmod, although most of the official rules and modules do.

## Links

- [Manage external dependencies with Bzlmod](https://bazel.build/build/bzlmod)
- [Bazel Migration Guide](https://docs.google.com/document/d/1JtXIVnXyFZ4bmbiBCr5gsTH4-opZAFf5DMMb-54kES0/edit?usp=sharing)
- [Bazel Central Registry](https://registry.bazel.build/)

**Comment by nlohmann** [comment]
This has since been implemented: the repository now ships a `MODULE.bazel` file at the root, so Bzlmod-based dependency management is already supported.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
